### ISSUE TITLE
Antispam: Posts flagged by Akismet need to be restored by user after admin approval

### ISSUE THREAD

**Issue by dcsjapan** [flarum/core#470]
Based on a Gitter conversation with user "lkhrs" (posting in the forums as Cactus):

When a moderator clears a post that Akismet has flagged as spam, it appears to the user who posted it as though it was deleted by that user. (At least, that's what it sounds like is happening.) The user has to Restore it to make it visible.

I can't say whether other non-moderator users are also seeing the post as deleted. If that's true, this bug could be further delaying contributions that have already been delayed once by an Akismet false positive.

Additional thoughts on the subject:

It might be a good idea to add a setting (if one doesn't already exist) to allow admins to configure what happens to posts flagged by Akismet, as follows:
- The post is reported and withheld for moderator approval (current behavior)
- The post is reported by displayed as normal

The latter setting would be useful in cases where Akismet is giving too many false positives and/or the forum moderators don't have the capacity to react quickly to reports.

**Comment by tobscure** [comment]
> When a moderator clears a post that Akismet has flagged as spam, it appears to the user who posted it as though it was deleted by that user. (At least, that's what it sounds like is happening.) The user has to Restore it to make it visible.

FYI, I can't reproduce this. I just clicked "Not Spam" on [this post](http://discuss.flarum.org/d/636-help-me-set-up-my-conf-file/30), and it was restored without any further action.

> It might be a good idea to add a setting (if one doesn't already exist) to allow admins to configure what happens to posts flagged by Akismet.

This is a good idea.

**Comment by dcsjapan** [comment]
> FYI, I can't reproduce this. I just clicked "Not Spam" on this post, and it was restored without any further action.

Yes, I've noticed the same thing while clearing several of his posts. They're restored for me as expected, but he's reporting that he has to Restore them on his end. Very odd.

Perhaps you could try posting something spam-like about Viagra and whatnot in the Sandbox, and I can clear it for you ... Maybe that would do it?

**Comment by tobscure** [comment]
@dcsjapan done, let's try this out. http://discuss.flarum.org/d/1011-this-is-test-spam

**Comment by dcsjapan** [comment]
Okay, not a bug. If I catch Cactus in chat, I'll ask him to let me know if it happens again, and try to get more specific info about what's happening. Thanks for looking into it!

**Comment by tobscure** [comment]
Just making a note to make a deleted post collapse when the "ignore" button is pressed on a report.

**Comment by dcsjapan** [comment]
This probably belongs as a new issue, but we've had a [request from a user](http://discuss.flarum.org/d/1033-posts-held-for-moderation-should-positively-indicate-status) for more positive indication of the status of spam flagged by Akisment. This strikes me as a good idea, since I've noticed more duplicate posts (as a result of false positives) since Akismet was put in place.

It might be good to consider this angle in connection with my suggestion re: admin configuration, above.
